crates/nu-command/src/filesystem/ucp.rs

```diff
@@ -109,10 +111,19 @@ impl Command for UCp {
         _input: PipelineData,
     ) -> Result<PipelineData, ShellError> {
         let interactive = call.has_flag(engine_state, stack, "interactive")?;
-        let (update, copy_mode) = if call.has_flag(engine_state, stack, "update")? {
-            (UpdateMode::ReplaceIfOlder, CopyMode::Update)
+        let update = if call.has_flag(engine_state, stack, "update")? {
+            UpdateMode::ReplaceIfOlder
         } else {
-            (UpdateMode::ReplaceAll, CopyMode::Copy)
+            UpdateMode::ReplaceAll
+        };
+        let copy_mode = if call.has_flag(engine_state, stack, "link")? {
+            CopyMode::Link
+        } else if call.has_flag(engine_state, stack, "symbolic-link")? {
+            CopyMode::SymLink
+        } else if call.has_flag(engine_state, stack, "update")? {
+            CopyMode::Update
+        } else {
+            CopyMode::Copy
```

Comment:
I've tried to directly port the flag behavior from the uu_cp crate:
https://github.com/uutils/coreutils/blob/7e4ef50c82a7503cba05b97c1c37970b9b4c8346/src/uu/cp/src/cp.rs#L760-L782

Reply:
My opinion is that, if nushell's cp continues to implement more flags with passthrough to uutils cp, then we should find a way to try and delegate the above sorts of switch logic to uutils's cp instead of reimplementing it. Copying the switch logic from uutils to nushell's codebase is more error prone and requires writing more tests.